src/error.rs

```diff
@@ -2,44 +2,128 @@ use raw_window_handle::{RawDisplayHandle, RawWindowHandle};
         window_handle: RawWindowHandle,
     },
 
-    #[error("The provided window handle is null.")]
+    /// The [`RawWindowHandle`] passed into [`Surface::new`] is missing necessary fields.
+    ///
+    /// [`RawWindowHandle`]: raw_window_handle::RawWindowHandle
+    /// [`Surface::new`]: crate::Surface::new
     IncompleteWindowHandle,
 
-    #[error("The provided display handle is null.")]
+    /// The [`RawDisplayHandle`] passed into [`Context::new`] is missing necessary fields.
+    ///
+    /// [`RawDisplayHandle`]: raw_window_handle::RawDisplayHandle
+    /// [`Context::new`]: crate::Context::new
     IncompleteDisplayHandle,
 
-    #[error("Surface size {width}x{height} out of range for backend.")]
+    /// The provided size is outside of the range supported by the backend.
     SizeOutOfRange {
+        /// The width that was out of range.
         width: NonZeroU32,
+
+        /// The height that was out of range.
         height: NonZeroU32,
     },
 
-    #[error("Platform error")]
+    /// A platform-specific backend error occurred.
+    ///
+    /// The first field provides a human-readable description of the error. The second field
+    /// provides the actual error that occurred. Note that the second field is, under the hood,
+    /// a private wrapper around the actual error, preventing the user from downcasting to the
+    /// actual error type.
     PlatformError(Option<String>, Option<Box<dyn Error>>),
 }
 
+impl fmt::Display for SoftBufferError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::UnsupportedDisplayPlatform {
+                human_readable_display_platform_name,
+                display_handle,
+            } => write!(
+                f,
+                "The provided display returned an unsupported platform: {}.\nDisplay handle: {:?}",
+                human_readable_display_platform_name, display_handle
+            ),
+            Self::UnsupportedWindowPlatform {
+                human_readable_window_platform_name,
+                human_readable_display_platform_name,
+                window_handle,
+            } => write!(
+                f,
+                "The provided window returned an unsupported platform: {}, {}.\nWindow handle: {:?}",
+                human_readable_window_platform_name, human_readable_display_platform_name, window_handle
+            ),
+            Self::IncompleteWindowHandle => write!(f, "The provided window handle is null."),
+            Self::IncompleteDisplayHandle => write!(f, "The provided display handle is null."),
+            Self::SizeOutOfRange { width, height } => write!(
+                f,
+                "Surface size {}x{} out of range for backend.",
+                width, height
+            ),
+            Self::PlatformError(msg, None) => write!(f, "Platform error: {:?}", msg),
+            Self::PlatformError(msg, Some(err)) => write!(f, "Platform error: {:?}: {}", msg, err),
+        }
```

Comment:
We can already use implicit named arguments (`format!("{msg}")`). Was stabilized in 1.58, our MSRV is 1.64.

Reply:
I used implicit named arguments for the shorter names. I think that the longer names shouldn't be embedded into the format string.